operator/internal/manifests/var.go

```diff
@@ -664,3 +664,39 @@ func getInstanceAvailabilityZoneEnvVar() corev1.EnvVar {
 		},
 	}
 }
+
+func initContainerZoneAnnotationCheck(image string) corev1.Container {
+	return corev1.Container{
+		Name:  "zone-annotation-check",
+		Image: image,
+		Command: []string{
+			"sh",
+			"-c",
+			"while ! [ -s /etc/zone-annotation/annotation ]; do echo Waiting for zone annotation to be set; sleep 2; done; echo Zone annotation is set; cat /etc/zone-annotation/annotation",
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "zone-annotation",
+				MountPath: "/etc/zone-annotation",
+			},
+		},
+	}
+}
+
+func zoneAnnotationVolumeMount() corev1.Volume {
+	return corev1.Volume{
+		Name: "zone-annotation",
+		VolumeSource: corev1.VolumeSource{
+			DownwardAPI: &corev1.DownwardAPIVolumeSource{
+				Items: []corev1.DownwardAPIVolumeFile{
+					{
+						Path: "annotation",
+						FieldRef: &corev1.ObjectFieldSelector{
+							FieldPath: "metadata.annotations['loki_instance_availability_zone']",
```

Comment:
Let's a keep a note to move the value of this field to a constant once the parent PR is merged if not done there already.